Client/Frontend/Browser/Tab Management/TabManagerStore.swift

```diff
@@ -141,39 +132,62 @@ class TabManagerStoreImplementation: TabManagerStore, FeatureFlaggable, Loggable
         }
 
         var tabToSelect: Tab?
-
         for savedTab in savedTabs {
             // Provide an empty request to prevent a new tab from loading the home screen
-            var tab = tabManager.addTab(flushToDisk: false, zombie: true, isPrivate: savedTab.isPrivate)
+            var tab = addTabClosure(savedTab.isPrivate)
             tab = savedTab.configureSavedTabUsing(tab, imageStore: imageStore)
             if savedTab.isSelected {
                 tabToSelect = tab
             }
         }
 
-        if tabToSelect == nil {
-            tabToSelect = tabManager.tabs.first(where: { $0.isPrivate == false })
-        }
-
         return tabToSelect
     }
 
     func clearArchive() {
         guard let path = tabsStateArchivePath() else { return }
-        try? FileManager.default.removeItem(atPath: path)
+        do {
+            try fileManager.removeItem(atPath: path)
+        } catch let error {
+            browserLog.warning("Clear archive couldn't be completed with error: \(error.localizedDescription)")
```

Comment:
Wondering if we should log this in sentry as well 🤔

Reply:
Good point, I can make the change in my subsequent PR where I want to fix the deprecation warning